contracts/token/ERC721/ERC721.sol

```diff
@@ -333,11 +336,11 @@ contract ERC721 is Context, ERC165, IERC721, IERC721Metadata {
         address to,
         uint256 tokenId
     ) internal virtual {
-        require(ERC721.ownerOf(tokenId) == from, "ERC721: transfer from incorrect owner");
```

Comment:
It is wrong to remove this from here... We need to check that `from` is the owner at the time of the transfer. We should do the same we did in `_burn`, and re-read the owner after the hook...

Reply:
My impression was that it would be checked after the hook, and that would revert in the owner is not correct.

The risk is that the hook would be executed with an invalid `from`, and the revert would come after. 

There is a world in which the from is not valid, the before hook move the token from its current owner to the from, and then the test passes and we do what we have to do... The code would have to be really f****d up, but its possible.

My concern is that doing the test twice increase transfer cost by >100 gas (on probably the most common and scrutinized operation)